### PR TITLE
Inherit module rp

### PR DESCRIPTION
**Pull request by perryr16**
Merged Inheritance
Gamable module has several helpers that were used more than once
Winable is an instance module. the method inside that used several times

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -1,9 +1,15 @@
-require "./lib/hashable"
+require_relative "hashable"
+require_relative "gameable"
+require_relative "winable"
+
```
Comment (perryr16) [inline lib/game.rb:4]: modules

File: lib/game.rb [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
 require_relative 'collection'
 
 class Game < Collection
 
+  include Winable
   extend Hashable
+  extend Gameable
+
```
Comment (perryr16) [inline lib/game.rb:12]: include for instance modules
exctend for class modules

File: lib/game.rb [hunk 3]
```diff
@@ -36,27 +41,13 @@ def self.average_goals_per_game
...
-      keys_to_string(average_goals_by_season)
-    end
 
-  def self.games_goals_by(hoa_team)
-    #{away_team_id => {goals => x, games_played => y}}
-    if hoa_team == :away_team
-      hash_of_hashes(all, :away_team_id, :goals, :games_played, :away_goals, 1)
-    elsif hoa_team == :home_team
-      hash_of_hashes(all, :home_team_id, :goals, :games_played, :home_goals, 1)
-    end
```
Comment (perryr16) [inline lib/game.rb:58]: boom! roasted

File: lib/gameable.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  #MODULE! helper used once (but child method is used 4 more times)
+    def games_goals_by(hoa_team)
+      #{away_team_id => {goals => x, games_played => y}}
+      if hoa_team == :away_team
+        hash_of_hashes(all, :away_team_id, :goals, :games_played, :away_goals, 1)
+      elsif hoa_team == :home_team
+        hash_of_hashes(all, :home_team_id, :goals, :games_played, :home_goals, 1)
+      end
+    end
+
```
Comment (perryr16) [inline lib/gameable.rb:33]: removed self. from methods to work as class modules

File: lib/winable.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+module Winable
+
+  #for use in Game
+  def win?(team_id)
+    away_win = team_id == @away_team_id && @away_goals > @home_goals
+    home_win =  team_id == @home_team_id && @home_goals > @away_goals
+    return 1 if away_win || home_win
+    0
+  end
```
Comment (perryr16) [inline lib/winable.rb:9]: this determines a "Win" and returns 1 if true, 0 if false. Ties return 0

**Review by perryr16 — COMMENTED** [review]
rp comments

File: lib/game.rb (hunk 1 above)
Comment (foleymichelle9) [reply lib/game.rb:4]: you love to see it!

File: lib/game.rb [hunk 6]
```diff
@@ -77,24 +68,6 @@ def self.lowest_scoring_visitor_team_id
...
-  def self.games_and_wins_by_season(team_id)
-      #{ season => {:wins => x, :games_played => y}}
-    hash_of_hashes(games_played_by(team_id), :season, :wins, :games_played, :win?, 1, team_id)
-  end
-
-  def self.win_percent_by_season(team_id)
-    # :wins / :games_played * 100
-    win_percent_by_season = divide_hash_values(:wins, :games_played, games_and_wins_by_season(team_id))
-    win_percent_by_season.transform_values { |v| (v * 100).to_i}
-  end
```
Comment (Rostammahabadi) [inline lib/game.rb:97]: this is awesome!

**Comment by Rostammahabadi** [comment]
looks great!

File: lib/winable.rb (hunk 5 above)
Comment (foleymichelle9) [reply lib/winable.rb:9]: nice!

**Comment by foleymichelle9** [comment]
Agreed! I approve merging :)

File: lib/gameable.rb (hunk 4 above)
Comment (stellakunzang) [reply lib/gameable.rb:33]: nice!

**Pull request merged**